com.unity.render-pipelines.high-definition/Editor/Material/ShaderGraph/HDShaderPasses.cs

```diff
@@ -280,22 +280,32 @@ RenderStateCollection GenerateRenderState()
 
             FieldCollection GenerateRequiredFields()
             {
-                return new FieldCollection()
+                if (supportLighting)
                 {
-                    HDStructFields.AttributesMesh.normalOS,
-                    HDStructFields.AttributesMesh.tangentOS,
-                    HDStructFields.AttributesMesh.uv0,
-                    HDStructFields.AttributesMesh.uv1,
-                    HDStructFields.AttributesMesh.color,
-                    HDStructFields.AttributesMesh.uv2,
-                    HDStructFields.AttributesMesh.uv3,
-                    HDStructFields.FragInputs.tangentToWorld,
-                    HDStructFields.FragInputs.positionRWS,
-                    HDStructFields.FragInputs.texCoord1,
-                    HDStructFields.FragInputs.texCoord2,
-                    HDStructFields.FragInputs.texCoord3,
-                    HDStructFields.FragInputs.color,
-                };
+                    return new FieldCollection()
+                    {
+                        HDStructFields.AttributesMesh.normalOS,
+                        HDStructFields.AttributesMesh.tangentOS,
+                        HDStructFields.AttributesMesh.uv0,
+                        HDStructFields.AttributesMesh.uv1,
+                        HDStructFields.AttributesMesh.color,
+                        HDStructFields.AttributesMesh.uv2,
+                        HDStructFields.AttributesMesh.uv3,
+                        HDStructFields.FragInputs.tangentToWorld,
+                        HDStructFields.FragInputs.positionRWS,
+                        HDStructFields.FragInputs.texCoord1,
+                        HDStructFields.FragInputs.texCoord2,
+                        HDStructFields.FragInputs.texCoord3,
+                        HDStructFields.FragInputs.color,
```

Comment:
I am not sure either, this is how it was done for HDRP prior to the PR and the modifications here were simply necessary to handle the Unlit case where the compiler optimization caused z-fighting artifacts.

If the PR is green for all tests I suggest we investigate this in a future PR since it is somewhat orthogonal to VFX's refactor, imo. cc @sebastienlagarde